File: common/src/main/java/dev/number6/db/dagger/DatabaseConfigurationPortModule.java
package dev.number6.db.dagger;

import dev.number6.db.port.DatabaseConfigurationPort;
import dev.number6.db.adaptor.DatabaseServiceConfigurationAdaptor;
import dagger.Module;
import dagger.Provides;

@Module
public class DatabaseConfigurationPortModule {

    @Provides
    DatabaseConfigurationPort dbConfig() {
        return new DatabaseServiceConfigurationAdaptor();
    }
}

File: comprehend/src/main/kotlin/dev/number6/db/dagger/DatabaseConfigurationPortModule.kt
package dev.number6.db.dagger

import dagger.Module
import dagger.Provides
import dev.number6.db.adaptor.DatabaseServiceConfigurationAdaptor
import dev.number6.db.port.DatabaseConfigurationPort

@Module
class DatabaseConfigurationPortModule {
    @Provides
    fun dbConfig(): DatabaseConfigurationPort {
        return DatabaseServiceConfigurationAdaptor()
    }
}